main/springboot/src/main/java/org/mobilitydata/gtfsvalidator/springboot/GtfsValidatorController.java

```diff
@@ -0,0 +1,129 @@
+      @RequestParam() String url,
+      @RequestParam(required = false, defaultValue = Arguments.VALIDATION_REPORT_DEFAULT_NAME_JSON)
+          String validation_report_name,
+      @RequestParam(required = false, defaultValue = Arguments.SYSTEM_ERRORS_REPORT_DEFAULT_NAME)
+          String system_error_report_name,
+      @RequestParam(required = false, defaultValue = "dataset id value") String dataset_id,
+      @RequestParam(required = false, defaultValue = "commit sha value") String commit_sha) {
+
+    final String[] argv = {
+      "-o", output_base,
+      "-t", threads,
+      "-c", country_code,
+      "-u", url,
+      "-v", validation_report_name,
+      "-e", system_error_report_name
+    };
+    Arguments args = new Arguments();
+    JCommander jCommander = new JCommander(args);
+    jCommander.parse(argv);
+    Main.main(argv);
+    return pushValidationReportToCloudStorage(commit_sha, dataset_id, args);
+  }
+
+  /**
+   * Pushes validation report to google cloud storage. Returns the status of the validation and
+   * validation report storage. Requires authentification to be set prior execution i.e. environment
+   * variables GOOGLE_APPLICATION_CREDENTIALS and VALIDATION_REPORT_BUCKET have to be defined.
+   *
+   * @param commitSha the commit SHA
+   * @param datasetId the id of the dataset
+   * @param args the {@code Argument} generated from the query parameters
+   * @return the status of the validation and validation report storage
+   */
+  private String pushValidationReportToCloudStorage(
+      String commitSha, String datasetId, Arguments args) {
+    // Instantiates a client
+    Storage storage = StorageOptions.getDefaultInstance().getService();
+    StringBuilder builder = new StringBuilder();
+    JsonObject root = new JsonObject();
+    JsonObject properties = new JsonObject();
+    root.add("properties", properties);
+
+    Bucket commitBucket =
+        storage.get(System.getenv(VALIDATION_REPORT_BUCKET), Storage.BucketGetOption.fields());
+    if (commitBucket == null) {
+      commitBucket =
+          storage.create(
+              BucketInfo.newBuilder(commitSha)
+                  .setStorageClass(StorageClass.STANDARD)
+                  .setLocation("US")
+                  .build());
+      logger.atInfo().log("Bucket %s created.", commitBucket.getName());
+    }
+    BlobId blobId =
+        BlobId.of(
+            commitBucket.getName(),
+            String.format("%s/%s/%s", commitSha, datasetId, args.getValidationReportName()));
+    BlobInfo blobInfo = BlobInfo.newBuilder(blobId).build();
+    try {
+      storage.create(
+          blobInfo,
+          Files.readAllBytes(
+              Paths.get(
+                  String.format("%s/%s", args.getOutputBase(), args.getValidationReportName()))));
+      builder.append("Execution fo the validator was successful.");
+    } catch (IOException ioException) {
```

Comment:
> Related to status codes - does the Google Cloud SDK provide more precise exceptions for unauthorized, etc.?

Yes, that is the case. 

> If so I'd suggest catching these separately (or using SDK methods to check explicitly first, if that's supported) so that you can return status codes specific to unauthorized, network problem, etc.

Modified.